src/CoreBundle/private/library/classes/dbobjects/TCMSDownloadFileEndPoint.class.php

```diff
@@ -160,46 +163,58 @@ public function &GetFileType()
     {
-        if (!array_key_exists('sFullURL', $this->_cacheParameter)) {
-            $sLink = $this->GetPlainDownloadLink($dummyLink, $bCreateLinkWithToken, false, $oExtranetUser);
+        return $this->getDownloadHtmlTag($dummyLink, $bHideName, $bHideSize, $bHideIcon, $sDownloadLinkName);
+    }
 
-            $mediaIdTag = '';
-            if ($dummyLink) {
-                $mediaIdTag = ' cmsdocument="'.$this->id.'"';
-            }
+    /**
+     * Renders HTML tag of download with icon and size.
+     */
+    public function getDownloadHtmlTag(
+        bool $isWysiwygBackendLink = false,
+        bool $hideName = false,
+        bool $hideSize = false,
+        bool $hideIcon = false,
+        string $downloadLinkName = ''): string
+    {
+        if (array_key_exists('downloadHtmlTag', $this->_cacheParameter)) {
+            return $this->_cacheParameter['downloadHtmlTag'];
+        }
 
-            $humanReadableFileSize = self::GetHumanReadableFileSize($this->sqlData['filesize']);
+        $viewRenderer = new \ViewRenderer();
 
-            $sName = TGlobalBase::OutHTML($this->GetName());
-            if (!empty($sDownloadLinkName)) {
-                $sName = TGlobal::OutHTML($sDownloadLinkName, false);
-            }
+        $downloadUrl = $this->GetPlainDownloadLink($isWysiwygBackendLink);
 
-            $extraStyle = '';
-            if ($bHideName) {
-                $sName = '&nbsp;';
-                $extraStyle .= 'style="text-decoration:none"';
-            }
+        $fileName = TGlobalBase::OutHTML($this->GetName());
+        if ('' !== $downloadLinkName) {
+            $fileName = TGlobal::OutHTML($downloadLinkName, false);
+        }
 
-            $sSizeHTML = '';
-            if (!$bHideSize) {
-                $sSizeHTML = '<span class="downloadsize">['.TGlobalBase::OutHTML($humanReadableFileSize).']</span>';
-            }
+        $downloadLinkDataModel = new DownloadLink($this->id, $downloadUrl, $fileName);
+        $downloadLinkDataModel->setIsBackendLink($isWysiwygBackendLink);
+        $downloadLinkDataModel->setHumanReadableFileSize(self::GetHumanReadableFileSize($this->sqlData['filesize']));
+        $downloadLinkDataModel->setShowSize(!$hideSize);
 
-            if ($bHideIcon) {
-                $sIconClass = 'nofileicon';
-            } else {
-                $oFileType = &$this->GetFileType();
-                $sIconClass = TGlobalBase::OutHTML($oFileType->sqlData['file_extension']);
-            }
-            $url = "<span class=\"cmsdownloaditem\"{$mediaIdTag}><a {$extraStyle} class=\"".$sIconClass."\" target=\"_blank\" href=\"{$sLink}\"
-        onclick=\"window.open('{$sLink}','downloaddetail','width=600,height=600,directories=no,location=no,menubar=no,resizable=yes,status=no,toolbar=no,scrollbars=yes'); return false;\">{$sName}</a>
-        ".$sSizeHTML.'</span>';
+        if (true === $hideName) {
```

Comment:
Both blocks look mostly like display logic? ($hideName, $hideIcon)

Can they be easily overwritten/restyled with the Twig/Css?

Reply:
The twig allows you to change the output completely, so yes.